services/storehost/extStats.go

```diff
@@ -0,0 +1,455 @@
+	reportInterval.Store(defaultReportInterval)
+}
+
+// ExtStatsReporterPause pauses the reporting (intended for tests)
+func ExtStatsReporterPause() {
+
+	swapped := atomic.CompareAndSwapInt32(&extStatsReporterPaused, 0, 1)
+	common.GetDefaultLogger().WithField(`was-running`, swapped).Info("extStatsReporter: paused")
+}
+
+// ExtStatsReporterResume resumes the reporting (intended for tests)
+func ExtStatsReporterResume() {
+
+	swapped := atomic.CompareAndSwapInt32(&extStatsReporterPaused, 1, 0)
+	common.GetDefaultLogger().WithField(`was-running`, !swapped).Info("extStatsReporter: resumed")
+}
+
+// ExtStatsReporterSetReportInterval updates the report interval (intended for tests)
+func ExtStatsReporterSetReportInterval(interval time.Duration) {
+
+	reportInterval.Store(interval)
+	common.GetDefaultLogger().WithField(`interval`, interval).Info("extStatsReporter: updated report interval")
+}
+
+func NewExtStatsReporter(hostID string, xMgr *ExtentManager, mClient metadata.TChanMetadataService, logger bark.Logger) *ExtStatsReporter {
+
+	return &ExtStatsReporter{
+		hostID:  hostID,
+		xMgr:    xMgr,
+		mClient: mClient,
+		logger:  logger,
+
+		extents: extStatsContextMap{
+			xmap: make(map[string]*extStatsContext),
+		},
+
+		reportPool: sync.Pool{New: func() interface{} { return &report{} }},
+		reportC:    make(chan *report, reportChanBufLen),
+		stopC:      make(chan struct{}),
+	}
+}
+
+func (t *ExtStatsReporter) Start() {
+
+	t.logger.Info("extStatsReporter: started")
+
+	// register callbacks to get notified every time an extent is opened/closed
+	t.xMgr.RegisterCallbacks(t)
+
+	t.wg.Add(1)
+	go t.reporterPump()
+
+	t.wg.Add(1)
+	go t.schedulerPump()
+}
+
+func (t *ExtStatsReporter) Stop() {
+
+	close(t.stopC)
+	t.wg.Wait()
+
+	t.logger.Info("extStatsReporter: stopped")
+}
+
+func (t *ExtStatsReporter) trySendReport(extentID uuid.UUID, ext *extentContext, ctx *extStatsContext) bool {
+
+	// if paused, do nothing
+	if atomic.LoadInt32(&extStatsReporterPaused) == 1 {
+		return false
+	}
+
+	r := t.reportPool.Get().(*report)
+
+	r.ctx = ctx
+	r.extentID = extentID
+	r.timestamp = time.Now().UnixNano()
+
+	// get a snapshot of various extent stats
+	r.firstSeqNum, r.lastSeqNum = ext.getBeginSeqNum(), ext.getLastSeqNum()
+	r.sealed, _ = ext.isSealed()
+
+	// create and send report non-blockingly
+	select {
+	case t.reportC <- r:
+
+		return true // sent
+
+	default:
+		// drop the report, if we are running behind
+		if ctx.lastReport != nil {
+			t.reportPool.Put(ctx.lastReport) // return old 'lastReport' to pool
+		}
+		ctx.lastReport = r // update lastReport to this one
+
+		// t.logger.WithFields(bark.Fields{ // #perfdisable
+		// 	common.TagExt:  extentID,      // #perfdisable
+		// 	`first-seqnum`: r.firstSeqNum, // #perfdisable
+		// 	`last-seqnum`:  r.lastSeqNum,  // #perfdisable
+		// }).Info("extStatsReporter: report dropped") // #perfdisable
+
+		return false // not sent
+	}
+}
+
+func (t *ExtStatsReporter) schedulerPump() {
```

Comment:
this method is a great place to emit a metric that gives the count of mutated extents during one time slot. 
Also, would be cool to cool to emit other latency/count metrics from extStats